### ISSUE TITLE
Combine integer flags more readable

### ISSUE THREAD

**Issue by programadorthi** [android/android-ktx#254]
Original post link below:

[Best way to combine integer flags using Kotlin?](https://stackoverflow.com/questions/34004977/best-way-to-combine-integer-flags-using-kotlin)

**Comment by simonschiller** [comment]
Are you proposing to introduce the `with` infix function from the post? I think the naming is not ideal, as it could be confused with the `with` high order function from the `kotlin-stdlib`.
Also this does not belong in this library as it is not Android specific.

**Comment by JakeWharton** [comment]
Combining flags comes from the language API and we don't want to invent a special Android-only version of it. Setting system UI flags more easily is covered by #139.

**Comment by programadorthi** [comment]
You have a good point. But the word `or` in plain English "mean": One or other. But in the programming it is used as a merge in bitwise operations. So, rename the infix operator to `merge` or `mergeWith` to be more readable.

```
window.decorView.systemUiVisibility =
    View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN merge
    View.SYSTEM_UI_FLAG_LAYOUT_STABLE merge
    View.SYSTEM_UI_FLAG_LOW_PROFILE
```

**Comment by JakeWharton** [comment]
No, "or" in English can mean both xor and inclusive-or depending on context. "Bring Jack or Jill" is xor. "You must be 12 years old or 4 feet tall to ride" is inclusive or.

**Comment by programadorthi** [comment]
Yes, you right!
